math/inline/linalg-raw.c

```diff
@@ -0,0 +1,289 @@
+#include <cblas.h>
+
+
+double kk_asum(kk_vector_t v, kk_context_t* ctx) {
```

Comment:
A few thoughts in general:

Maybe we should create a vector/blasvector and blasvector/vector inline conversion functions, so you don't have to do it all the time in these functions (represent the blasvector as just `value struct blasvector {internal: intptr_t}`). This makes a blasvector it it's own type, but essentially no overhead, since it is a value type, and just has an pointer under the hood (castable to double*). Then you can use vectors until you want high performance blas routines - then convert to blas representation, stay in that representation for awhile while you are using the api and just get a final `double` result at the end, or convert back to `vector<double>` at some point.

Reply:
I'll be sure to change it.

Reply:
How do I make it so that Koka will automatically clean up the blasvector's buffer?

Reply:
Ah you need my PR for automatic clean up based on the Koka wrapper's reference count: https://github.com/koka-lang/koka/pull/494/files

Especially `cextern.h`:
https://github.com/koka-lang/koka/pull/494/files#diff-78c2b9b5c063f1251aec07a6b1d2ec243fbcbb1a2db8ed188d62dd7711351dcc

The Koka bindings for that file are here: https://github.com/koka-lang/koka/pull/494/files#diff-40c7f5c96c37d8bac9614990b3a4d23e254e4369e9b9d2dc18c0ce8ef638550e

I'm pretty sure you could do this without changes to the compiler by using a polymorphic type `b` or something (which would translate to a boxed type, which is what needs to be the parameter type on the c side).

Reply:
Added to our std library here: https://github.com/koka-community/std/blob/main/std/cextern.kk
With example usage here: https://github.com/koka-community/std/blob/main/tests/cextern.kk